src/testhelpers.rs

```diff
@@ -38,7 +39,15 @@ pub enum TestEvents {
     Other,
 }
 
-impl EventData for TestEvents {}
+impl EventData for TestEvents {
+    fn event_type(&self) -> String {
+        match &self {
+            TestEvents::Inc(_) => "some_namespace.Inc",
+            TestEvents::Dec(_) => "some_namespace.Dec",
+            _ => panic!("Type for unhandled event not available"),
+        }.into()
+    }
+}
```

Comment:
What about this:

```rust
fn event_type(&self) -> String {
    #[derive(Deserialize)]
    struct EventTypeWrapper {
        #[serde(rename = "type")]
        event_type: String,
    }

    let event: EventTypeWrapper =
        from_value(to_value(self).expect("Encode failed")).expect("Decode failed");

    event.event_type
}
```

The (micro)benchmark sez it's 345% slower, but I'd rather not deal with strings directly and take the performance penalty.